python/pyarrow/tests/test_flight.py

```diff
@@ -180,20 +180,16 @@ def do_get(self, context, ticket):
     def do_put(self, context, descriptor, reader, writer):
         counter = 0
         expected_data = [-10, -5, 0, 5, 10]
-        while True:
-            try:
-                batch, buf = reader.read_chunk()
```

Comment:
Do we want to test both patterns (iterator vs calling read and catching StopIteration)?

Reply:
Same question!

Reply:
My initial thinking is no, since StopIteration must be called for an iterator to exit. Do you see value in adding both? I'm happy to add another test case if so.

Reply:
Ah I suppose an extra test would ensure that the underlying behavior of the iterator implementation doesn't change. Still, I lean towards omitting the additional test case, but do not feel strongly either way.